File: DSA/LeetCode/Stack/1910. Remove All Occurrences of a Substring.js
/**
 * @param {string} s
 * @param {string} part
 * @return {string}
 */
var removeOccurrences = function (s, part) {

    let stack = ""
    let pL = part.length
    for (let i = 0; i < s.length; i++) {
        stack += s[i]
        if (stack.length >= pL && stack.substring(stack.length - pL) == part) {
            stack = stack.slice(0, - pL)
        }
    }

    return stack


};

File: DSA/LeetCode/Stack/1910. Remove All Occurrences of a Substring.py
class Solution:
    def removeOccurrences(self, s: str, part: str) -> str:

        stack = ""
        pL = len(part)
        for a in s:
            stack += a

            if len(stack) >= pL and stack[-pL:] == part:
                stack = stack[:-pL]

        return stack
